python/tests/operations/split.py

```diff
@@ -0,0 +1,524 @@
+                names=["front and", "center"], values=np.array([[0, 1], [6, 7], [7, 4]])
+            ),
+            Labels(
+                names=["front and", "center"], values=np.array([[2, 4], [3, 3], [4, 7]])
+            ),
+            Labels(
+                names=["front and", "center"],
+                values=np.array([[1, 5], [5, 3], [8, 10]]),
+            ),
+        ]
+        with self.assertRaises(ValueError) as cm:
+            fn.split(self.tensor, axis="samples", grouped_idxs=grouped_idxs),
+        self.assertEqual(
+            str(cm.exception),
+            "a name passed in a Labels object at position 0 of ``grouped_idxs`` does"
+            + " not appear in the samples names of the input tensor",
+        )
+
+    def test_split_block_errors(self):
+        # TypeError not TB
+        with self.assertRaises(TypeError) as cm:
+            fn.split_block(self.tensor, axis="samples", grouped_idxs=self.grouped_idxs),
+        self.assertEqual(
+            str(cm.exception), "``block`` should be an equistore ``TensorBlock``"
+        )
+
+
+def _unique_indices(
+    tensor: Union[TensorMap, TensorBlock],
+    axis: str,
+    names: Union[List[str], str],
+):
+    """
+    For a given ``axis`` (either "samples" or "properties"), and for the given
+    samples/proeprties ``names``, returns a :py:class:`Labels` object of the
+    unique indices in the input ``tensor``, which can be either a
+    :py:class:`TensorMap` or :py:class:`TensorBlock` object.
+    """
+    # Parse tensor into a list of blocks
+    if isinstance(tensor, TensorMap):
+        blocks = tensor.blocks()
+    else:  # TensorBlock
+        blocks = [tensor]
+
+    # Parse the names into a list
+    names = [names] if isinstance(names, str) else names
+    names = list(names) if isinstance(names, tuple) else names
+
+    # Extract indices from each block
+    all_idxs = []
+    for block in blocks:
+        block_idxs = (
+            block.samples[names] if axis == "samples" else block.properties[names]
+        )
+        for idx in block_idxs:
+            all_idxs.append(idx)
+    if len(all_idxs) == 0:  # return an empty Labels, with the correct name
+        return Labels(names=names, values=np.array([[i for i in range(len(names))]]))[
+            :0
+        ]
+
+    # Define the unique indices, convert to a Labels obj, and return
+    unique_idxs = np.unique(all_idxs, axis=0)  # this also sorts the idxs
+    return Labels(names=names, values=np.array([[j for j in i] for i in unique_idxs]))
+
+
+def _searchable_labels(labels: Labels):
+    """
+    Returns the input Labels object but after being used to construct a
+    TensorBlock, so that look-ups can be performed.
+    """
+    return TensorBlock(
+        values=np.full((len(labels), 1), 0.0),
+        samples=labels,
+        components=[],
+        properties=Labels(["p"], np.array([[0]], dtype=np.int32)),
+    ).samples
+
+
+def _labels_equal(a: Labels, b: Labels, exact_order: bool):
```

Comment:
this function is very nice, I would put it in some _utilis file in src not hidden here

Reply:
but it should also check the label.names

Reply:
Do you mean in ``equistore/python/src/equistore/operations/_utils.py``?